backend/social/models.py

```diff
@@ -27,3 +28,29 @@ def add_message(sender, instance, created, **kwargs):
     from social.subscriptions import MessageSubscription
     if created:
         MessageSubscription.broadcast(payload={'message': instance}, group=f'group_{instance.group.id}')
+
+#Comprobar que un usuario no invie un mensaje a un grupo al que no pertenece
+@receiver(pre_save, sender=Message)
+def check_user_in_group(sender, instance, **kwargs):
+    if instance.user not in instance.group.users.all():
+        raise Exception('User not in group')
+
+
+# Comrpobar que un grupo individual solo tenga un usuario
+@receiver(pre_save, sender=Group)
+def check_individual_group(sender, instance, **kwargs):
+    if instance.individual:
+        if instance.users.count() > 1:
+            raise Exception('Individual group can only have one user')
+
+# No eliminar un grupos no individuales si tienen más de tres usuarios
+@receiver(pre_delete, sender=Group)
+def check_group(sender, instance, **kwargs):
+    if not instance.individual and instance.users.count() > 3:
```

Comment:
La validación está mal ya que debería ser `if not instance.individual and instance.users.count() > 2:` pues 3 es el mínimo para crear y mantener un grupo.
También haría falta validar `if instance.individual and instance.users.count() == 1` para poder eliminar los chats individuales. Esta validación se puede añadir a la anterior con un `or` o si se requiere de un mensaje más específico como otro `if`.